core/src/main/scala/org/apache/spark/sql/eventhubs/EventHubsSource.scala

```diff
@@ -204,11 +210,31 @@ private[spark] class EventHubsSource private[eventhubs] (sqlContext: SQLContext,
     // If not, we'll report possible data loss.
     earliestSeqNos = Some(earliestAndLatest.map {
       case (p, (e, _)) => NameAndPartition(ehName, p) -> e
-    }.toMap)
+    })
 
     val latest = earliestAndLatest.map {
       case (p, (_, l)) => NameAndPartition(ehName, p) -> l
-    }.toMap
+    }
+
+    val fromSeqNumbers = currentSeqNos match {
```

Comment:
Can you please explain why we are merging these two maps? Is the change for supporting a scenario where a new partition is added while a job is running?